File: 2010_2011_sujet_3_convergences/Markov.m
function [Y ] = Markov( X )
%UNTITLED2 Summary of this function goes here
%   Detailed explanation goes here
U=rand(1);
if (X==1) Y=2*(U<1/3) + 3*(U>1/3); end
if (X==2) Y=1*(U<1/3) + 3*(U>1/3); end
if (X==3) Y=1*(U<2/3) + 2*(U>2/3); end

end

File: 2012_2013_sujet_4_problemes_divers/Markov.py
import numpy as np


def Markov(X):
    U = np.random.rand()
    if X == 1:
        Y = 2 * (U < 1/3) + 3 * (U > 1/3)
    elif X == 2:
        Y = 1 * (U < 1/3) + 3 * (U > 1/3)
    elif X == 3:
        Y = 1 * (U < 2/3) + 2 * (U > 2/3)
    else:
        Y = X
    return int(Y)
